src/main/java/io/cdap/e2e/pages/locators/WranglerLocators.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.e2e.pages.locators;
+
+import io.cdap.e2e.utils.SeleniumDriver;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.How;
+
+/**
+ * CDF wrangler related locators.
+ */
+public class WranglerLocators {
+  @FindBy(how = How.XPATH, using = "//div[@data-testid='common-components-scrollableList-scrollDownButton']")
+  public static WebElement scrollButton;
+
+  @FindBy(how = How.XPATH, using = "//div[@data-testid='features-dataprep-directives-changeDataType-submenu-decimal" +
+    "-scaleInput']/input")
+  public static WebElement scaleText;
+
+  @FindBy(how = How.XPATH, using = "//button[contains(text(),'Extract')]")
+  public static WebElement extractButton;
```

Comment:
Do we have a 'data-testid' attribute for this element?

Reply:
yes we have but for every directive we have diff test id for extract button so we are simple going with 1 contains text rather than  for eg-15 diff extract buttons.